lib/ae_mdw/db/mutations/stats_mutation.ex

```diff
@@ -4,79 +4,115 @@ defmodule AeMdw.Db.StatsMutation do
   """
 
   alias AeMdw.Blocks
+  alias AeMdw.Db.IntTransfer
   alias AeMdw.Db.Model
   alias AeMdw.Db.Util
   alias AeMdw.Ets
   alias AeMdw.Mnesia
 
   require Model
 
-  defstruct [:height, :prev_sum_stat, :token_supply_delta]
+  defstruct [:stat, :sum_stat]
 
   @opaque t() :: %__MODULE__{
-            height: Blocks.height(),
-            prev_sum_stat: Model.sum_stat(),
-            token_supply_delta: integer()
+            stat: Model.stat(),
+            sum_stat: Model.sum_stat()
           }
 
-  @spec new(Blocks.height()) :: t()
-  def new(height) do
-    token_supply_delta = AeMdw.Node.token_supply_delta(height + 1)
-
-    prev_sum_stat = get_sum_stat(height)
+  @spec new(Blocks.height(), boolean()) :: t()
+  def new(height, all_cached?) do
+    m_stat = make_stat(height + 1, all_cached?)
+    Model.stat(block_reward: inc_block_reward, dev_reward: inc_dev_reward) = m_stat
+    m_sum_stat = make_sum_stat(height + 1, inc_block_reward, inc_dev_reward)
```

Comment:
Can we instead, move these computations outside of the `new` function and into some private function in the `Transaction` module? This way, `new` function only does some very simple struct building. Mutations should only be have the logic for writing to mnesia on the `mutate` function

Reply:
Sure, for a shallower and lighter `new`, a factory was added as requested.
Regarding the writing, this PR already moves reading logic out of the `mutate/1` leaving it to perform only `Mnesia` writes as by design.